test/Table_Tests/src/IO/Fetch_Spec.enso

```diff
@@ -94,3 +103,330 @@ add_specs suite_builder =
+        with_counts ~action =
+            before_count = get_num_response_cache_entries
+            action
+            after_count = get_num_response_cache_entries
+            [before_count, after_count]
+
+        reset_size_limits =
+            EnsoHTTPResponseCache.clearMaxFileSizeOverrideTestOnly
+            EnsoHTTPResponseCache.clearMaxTotalCacheSizeOverrideTestOnly
+
+        expect_counts expected_counts ~action =
+            counts = with_counts action
+            counts . should_equal expected_counts frames_to_skip=1
+
+        get_cache_file_sizes : Vector Integer
+        get_cache_file_sizes -> Vector Integer =
+            Vector.from_polyglot_array EnsoHTTPResponseCache.getFileSizesTestOnly . sort Sort_Direction.Ascending
+
+        url0 = base_url_with_slash+'test_download?max-age=16&length=10'
+        url1 = base_url_with_slash+'test_download?max-age=16&length=20'
+        url0_body = "aaaaaaaaaa"
+        url1_body = "aaaaaaaaaaaaaaaaaaaa"
+        url_post = base_url_with_slash + "post"
+        headers0 = [Header.new "A-Header" "a-header-value", Header.new "A-Header" "a-header-value"]
+        headers1 = [Header.new "A-Header" "a-different-header-value", Header.new "A-Header" "a-header-value"]
+
+        group_builder.specify "Cache should return the same repsonse" pending=pending_has_url <| Test.with_retries <|
+            HTTP.clear_response_cache
+
+            HTTP.fetch url0 . decode_as_text . should_equal url0_body
+            get_num_response_cache_entries . should_equal 1
+            HTTP.fetch url0 . decode_as_text . should_equal url0_body
+            get_num_response_cache_entries . should_equal 1
+            HTTP.fetch url1 . decode_as_text . should_equal url1_body
+            get_num_response_cache_entries . should_equal 2
+            HTTP.fetch url1 . decode_as_text . should_equal url1_body
+            get_num_response_cache_entries . should_equal 2
+
+            HTTP.clear_response_cache
+
+            HTTP.fetch url0 cache_policy=Cache_Policy.Use_Cache . decode_as_text . should_equal url0_body
+            HTTP.fetch url0 cache_policy=Cache_Policy.Use_Cache . decode_as_text . should_equal url0_body
+            HTTP.fetch url1 cache_policy=Cache_Policy.Use_Cache . decode_as_text . should_equal url1_body
+            HTTP.fetch url1 cache_policy=Cache_Policy.Use_Cache . decode_as_text . should_equal url1_body
+            get_num_response_cache_entries . should_equal 2
+
+            HTTP.clear_response_cache
+
+            HTTP.fetch url0 cache_policy=Cache_Policy.No_Cache . decode_as_text . should_equal url0_body
+            HTTP.fetch url0 cache_policy=Cache_Policy.No_Cache . decode_as_text . should_equal url0_body
+            HTTP.fetch url1 cache_policy=Cache_Policy.No_Cache . decode_as_text . should_equal url1_body
+            HTTP.fetch url1 cache_policy=Cache_Policy.No_Cache . decode_as_text . should_equal url1_body
+            get_num_response_cache_entries . should_equal 0
```

Comment:
What is the purpose of calling `fetch url0` in this test case? I don't see any difference from calling it just once.

Reply:
My suggestion would be - what if the `test_download` endpoint returned a random string each time? Then you could see if the response was cached between 2 invocations or not (if its the same response it must have been cached, because a new response would have differed).

Reply:
I'm fetching it multiple times so that the file count will tell me if it's using the cache. But agreed that it's better to randomize the data and check that it's different every time, so I did that too.